substra/sdk/client.py

```diff
@@ -721,6 +721,54 @@ def download_objective(self, key: str, destination_folder: str) -> None:
             os.path.join(destination_folder, 'metrics.py'),
         )
 
+    @logit
+    def download_model(self, key: str, folder) -> None:
+        """Download model to destination file."""
+        self._backend.download_model(key, os.path.join(folder, f'model_{key}'))
+
+    @logit
+    def download_traintuple_model(self, tuple_key: str, folder) -> None:
+        """Download traintuple model to destination file."""
+        self._download_tuple_model(schemas.Type.Traintuple, tuple_key, folder)
+
+    @logit
+    def download_aggregatetuple_model(self, tuple_key: str, folder) -> None:
+        """Download aggregatetuple model to destination file."""
+        self._download_tuple_model(schemas.Type.Aggregatetuple, tuple_key, folder)
+
+    @logit
+    def download_composite_traintuple_head_model(self, tuple_key: str, folder) -> None:
+        """Download composite traintuple head model to destination file."""
+        self._download_tuple_model(schemas.Type.CompositeTraintuple, tuple_key, folder, "head")
+
+    @logit
+    def download_composite_traintuple_trunk_model(self, tuple_key: str, folder) -> None:
+        """Download composite traintuple trunk model to destination file."""
+        self._download_tuple_model(schemas.Type.CompositeTraintuple, tuple_key, folder, "trunk")
+
+    def _download_tuple_model(self, tuple_type, tuple_key, folder, head_trunk=None) -> None:
+        """Download model to a destination file."""
+        tuple = self._backend.get(tuple_type, tuple_key)
```

Comment:
Can you check the status and throw an error if it is different from 'done' ? Otherwise for a traintuple still in progress for example, we only get an error `Traintuple 94ab89a3-b61f-44c0-bd78-e3392f5bf979 has no out-model`

Reply:
What about `Traintuple {key}, status {status} has no out-model.` ? Something along those lines, to make explicit the reason why it fails.

Reply:
Thanks for the suggestion, it looks great https://github.com/SubstraFoundation/substra/pull/261/commits/f485a5cf8ac070375ba937e82ed928c4008c2299